File: scripts/check_data.py
import sqlite3

conn = sqlite3.connect(r'D:/Claw/JAC_Year/jac_articles.db')
c = conn.cursor()

c.execute("SELECT count(*) FROM articles")
print(f"Total articles: {c.fetchone()[0]}")

c.execute('SELECT count(*) FROM articles WHERE abstract IS NOT NULL AND abstract != ""')
print(f"With abstract: {c.fetchone()[0]}")

c.execute('SELECT count(*) FROM articles WHERE keywords IS NOT NULL AND keywords != ""')
print(f"With keywords: {c.fetchone()[0]}")

c.execute('SELECT count(*) FROM articles WHERE citation IS NOT NULL AND citation != ""')
print(f"With citation: {c.fetchone()[0]}")

c.execute('SELECT count(*) FROM articles WHERE funding IS NOT NULL AND funding != ""')
print(f"With funding: {c.fetchone()[0]}")

c.execute('SELECT count(*) FROM articles WHERE corresponding_authors IS NOT NULL AND corresponding_authors != ""')
print(f"With corresponding_authors: {c.fetchone()[0]}")

c.execute('SELECT count(*) FROM articles WHERE institutions IS NOT NULL AND institutions != ""')
print(f"With institutions: {c.fetchone()[0]}")

c.execute('SELECT count(*) FROM articles WHERE received_date IS NOT NULL AND received_date != ""')
print(f"With received_date: {c.fetchone()[0]}")

c.execute('SELECT count(*) FROM articles WHERE accepted_date IS NOT NULL AND accepted_date != ""')
print(f"With accepted_date: {c.fetchone()[0]}")

c.execute('SELECT count(*) FROM articles WHERE published_date IS NOT NULL AND published_date != ""')
print(f"With published_date: {c.fetchone()[0]}")

c.execute('SELECT count(*) FROM articles WHERE affiliations IS NOT NULL AND affiliations != ""')
print(f"With affiliations: {c.fetchone()[0]}")

conn.close()

File: archive/root-cleanup-2026-05-19/_check_data.js
const initSqlJs = require('sql.js');
const fs = require('fs');

(async () => {
  const SQL = await initSqlJs();
  const db = new SQL.Database(fs.readFileSync('jac_articles.db'));
  
  // Check 2024 funding status more carefully
  const r2024 = db.exec(`SELECT 
    COUNT(*) as total,
    SUM(CASE WHEN funding IS NOT NULL AND funding != '' THEN 1 ELSE 0 END) as has_funding,
    SUM(CASE WHEN email IS NOT NULL AND email != '' THEN 1 ELSE 0 END) as has_email,
    SUM(CASE WHEN abstract IS NOT NULL AND abstract != '' THEN 1 ELSE 0 END) as has_abstract,
    SUM(CASE WHEN keywords IS NOT NULL AND keywords != '' THEN 1 ELSE 0 END) as has_keywords,
    SUM(CASE WHEN received_date IS NOT NULL AND received_date != '' THEN 1 ELSE 0 END) as has_received,
    SUM(CASE WHEN accepted_date IS NOT NULL AND accepted_date != '' THEN 1 ELSE 0 END) as has_accepted,
    SUM(CASE WHEN published_date IS NOT NULL AND published_date != '' THEN 1 ELSE 0 END) as has_published
    FROM articles WHERE year=2024`);
  
  const r2025 = db.exec(`SELECT 
    COUNT(*) as total,
    SUM(CASE WHEN funding IS NOT NULL AND funding != '' THEN 1 ELSE 0 END) as has_funding,
    SUM(CASE WHEN email IS NOT NULL AND email != '' THEN 1 ELSE 0 END) as has_email,
    SUM(CASE WHEN abstract IS NOT NULL AND abstract != '' THEN 1 ELSE 0 END) as has_abstract,
    SUM(CASE WHEN keywords IS NOT NULL AND keywords != '' THEN 1 ELSE 0 END) as has_keywords,
    SUM(CASE WHEN received_date IS NOT NULL AND received_date != '' THEN 1 ELSE 0 END) as has_received,
    SUM(CASE WHEN accepted_date IS NOT NULL AND accepted_date != '' THEN 1 ELSE 0 END) as has_accepted,
    SUM(CASE WHEN published_date IS NOT NULL AND published_date != '' THEN 1 ELSE 0 END) as has_published
    FROM articles WHERE year=2025`);
  
  console.log('=== 2024 (Volume 13) ===');
  console.log(r2024[0]);
  console.log('=== 2025 (Volume 14) ===');
  console.log(r2025[0]);
  
  db.close();
})();
